interfaces/builtin/ssh_public_keys.go

```diff
@@ -0,0 +1,50 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const sshPublicKeysSummary = `allows reading ssh public keys and non-sensitive configuration`
+
+const sshPublicKeysBaseDeclarationSlots = `
+  ssh-public-keys:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const sshPublicKeysConnectedPlugAppArmor = `
+# Description: Can read ssh public keys and non-sensitive configuration
+
+owner @{HOME}/.ssh/ r,
+owner @{HOME}/.ssh/environment r,
+owner @{HOME}/.ssh/*.pub r,
```

Comment:
Should this also allow reading `known_hosts`?

Reply:
> Should this also allow reading known_hosts?

No, public keys are public. The hosts you connect to is very private information. If we mix the two the interface becomes misleading and dangerous.